### PR TITLE
fixing flapMap signature

### PR DESCRIPTION
**Pull request by jerrypeng**

### CODE REVIEW

File: heron/api/src/java/com/twitter/heron/dsl/Streamlet.java [hunk 1]
```diff
@@ -85,7 +85,7 @@
    * flattening the result
    * @param flatMapFn The FlatMap Function that should be applied to each element
   */
-  <T> Streamlet<T> flatMap(SerializableFunction<? super R, Iterable<? extends T>> flatMapFn);
+  <T> Streamlet<T> flatMap(SerializableFunction<? super R, ? extends Iterable<T>> flatMapFn);
```
Comment (srkukarni) [inline heron/api/src/java/com/twitter/heron/dsl/Streamlet.java]: Would we rather have SerializableFunction<? super R, ? extends Iterable<? extends T>> flatMapFn

Comment (jerrypeng) [reply heron/api/src/java/com/twitter/heron/dsl/Streamlet.java]: What does that really buy us though?

Comment (jerrypeng) [reply heron/api/src/java/com/twitter/heron/dsl/Streamlet.java]: Ya we can change it to that

**Comment by jerrypeng** [comment]
@srkukarni thanks for the review! I have addressed your comments

**Pull request merged**